crates/lib/src/lib.rs

```diff
@@ -195,16 +196,38 @@ pub struct TypeAlias {
 #[derive(Debug, Clone, Eq, PartialEq, PartialOrd, Ord, de::Deserialize, ser::Serialize)]
 pub struct IndexDef {
     pub name: String,
-    pub ty: IndexType,
-    pub col_ids: Vec<u8>,
+    pub index_type: IndexType,
+    pub cols: Vec<u8>,
 }
 
-#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, de::Deserialize, ser::Serialize)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Display, de::Deserialize, ser::Serialize)]
 pub enum IndexType {
     BTree,
     Hash,
 }
 
+impl TryFrom<u8> for IndexType {
+    type Error = ();
+    fn try_from(v: u8) -> Result<Self, Self::Error> {
+        match v {
+            0 => Ok(IndexType::BTree),
+            1 => Ok(IndexType::Hash),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&str> for IndexType {
+    type Error = ();
+    fn try_from(v: &str) -> Result<Self, Self::Error> {
+        match v {
```

Comment:
I'm concerned about committing to the exact spelling here. Where did BTree vs btree come from?

Reply:
This is for `StIndexRow` where we need to parse the index type.